data-prepper-plugins/opensearch-api-source/src/test/java/org/opensearch/dataprepper/plugins/source/opensearchapi/OpenSearchAPISourceTest.java

```diff
@@ -0,0 +1,623 @@
+                                .contentType(MediaType.JSON_UTF_8)
+                                .build(),
+                        HttpData.ofUtf8(testBadData))
+                .aggregate()
+                .whenComplete((i, ex) -> assertSecureResponseWithStatusCode(i, HttpStatus.BAD_REQUEST)).join();
+
+        // Then
+        Assertions.assertTrue(testBuffer.isEmpty());
+        // Verify metrics
+        final Measurement requestReceivedCount = MetricsTestUtil.getMeasurementFromList(
+                requestsReceivedMeasurements, Statistic.COUNT);
+        Assertions.assertEquals(1.0, requestReceivedCount.getValue());
+        final Measurement badRequestsCount = MetricsTestUtil.getMeasurementFromList(
+                badRequestsMeasurements, Statistic.COUNT);
+        Assertions.assertEquals(1.0, badRequestsCount.getValue());
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    public void testBulkRequestJsonResponse400WithInvalidPayload(boolean includeIndexInPath) throws JsonProcessingException {
+        // Prepare
+        List<String> jsonList = new ArrayList<>();
+        for (int i = 0; i < 2; i++) {
+            jsonList.add(mapper.writeValueAsString(Collections.singletonMap("index", Collections.singletonMap("_index", "test-index"))));
+        }
+        final String testBadData = String.join("\n", jsonList);
+        openSearchAPISource.start(testBuffer);
+        refreshMeasurements();
+
+        // When
+        WebClient.of().execute(RequestHeaders.builder()
+                                .scheme(SessionProtocol.HTTP)
+                                .authority(AUTHORITY)
+                                .method(HttpMethod.POST)
+                                .path(includeIndexInPath ? "/" + TEST_INDEX + "/_bulk" : "/_bulk")
+                                .contentType(MediaType.JSON_UTF_8)
+                                .build(),
+                        HttpData.ofUtf8(testBadData))
+                .aggregate()
+                .whenComplete((i, ex) -> assertSecureResponseWithStatusCode(i, HttpStatus.BAD_REQUEST)).join();
+
+        // Then
+        Assertions.assertTrue(testBuffer.isEmpty());
+        // Verify metrics
+        final Measurement requestReceivedCount = MetricsTestUtil.getMeasurementFromList(
+                requestsReceivedMeasurements, Statistic.COUNT);
+        Assertions.assertEquals(1.0, requestReceivedCount.getValue());
+        final Measurement badRequestsCount = MetricsTestUtil.getMeasurementFromList(
+                badRequestsMeasurements, Statistic.COUNT);
+        Assertions.assertEquals(1.0, badRequestsCount.getValue());
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    public void testBulkRequestAPIJsonResponse413(boolean includeIndexInPath) throws JsonProcessingException {
+        testBulkRequestJsonResponse413(includeIndexInPath);
```

Comment:
Let's inline this. We don't need an extra method here.

Reply:
Thanks @dlvenable. I will resolve this.